From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Fri, 21 Aug 2015 20:17:46 +0000
Subject: Fix opening KRunner by typing on empty desktop
X-Git-Url: http://quickgit.kde.org/?p=plasma-workspace.git&a=commitdiff&h=d0469af1fb7ed39573d6e3c0eff048b1dbbb6289
---
Fix opening KRunner by typing on empty desktop

Let the key press event propagate first and if nobody accepted it,
we'll open KRunner

BUG: 347872
FIXED-IN: 5.4.1
REVIEW: 124871
---


--- a/shell/desktopview.cpp
+++ b/shell/desktopview.cpp
@@ -208,8 +208,10 @@
 
 void DesktopView::keyPressEvent(QKeyEvent *e)
 {
+    ContainmentView::keyPressEvent(e);
+
     // When a key is pressed on desktop when nothing else is active forward the key to krunner
-    if (!e->modifiers() && activeFocusItem() == contentItem()) {
+    if (!e->modifiers() && !e->isAccepted()) {
         const QString text = e->text().trimmed();
         if (!text.isEmpty() && text[0].isPrint()) {
             const QString interface("org.kde.krunner");
@@ -218,8 +220,6 @@
             e->accept();
         }
     }
-
-    ContainmentView::keyPressEvent(e);
 }
 
 
